sources/api/apiserver/src/server/controller.rs

```diff
@@ -420,21 +498,91 @@ pub(crate) fn get_metadata_for_data_keys<D: DataStore, S: AsRef<str>>(
     Ok(result)
 }
 
-/// Gets the value of a metadata key everywhere it's found in the data store.  Returns a mapping
-/// of data key to the metadata value associated with the requested key.
-pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+// Sort the given Hashmap in topological order
+fn get_sorted_metadata(
```

Comment:
nit: `get` in names usually implies a data read. I'd use a different name since this takes ownership.

```suggestion
fn topo_sort_metadata(
```

Reply:
Used `sort_metadata`